solr/core/src/java/org/apache/solr/update/processor/SkipExistingDocumentsProcessorFactory.java

```diff
@@ -0,0 +1,217 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.update.processor;
+
+import org.apache.lucene.util.BytesRef;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.core.SolrCore;
+import org.apache.solr.handler.component.RealTimeGetComponent;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+import org.apache.solr.search.SolrIndexSearcher;
+import org.apache.solr.update.AddUpdateCommand;
+import org.apache.solr.update.UpdateCommand;
+import org.apache.solr.util.RefCounted;
+import org.apache.solr.util.plugin.SolrCoreAware;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+
+import static org.apache.solr.common.SolrException.ErrorCode.SERVER_ERROR;
+import static org.apache.solr.update.processor.DistributingUpdateProcessorFactory.DISTRIB_UPDATE_PARAM;
+
+/**
+ * <p>
+ *     This Factory generates an UpdateProcessor that will (by default) skip inserting new documents
+ *     if there already exists a document with the same uniqueKey value in the index. It will also
+ *     skip Atomic Updates to a document if that document does not already exist. This behaviour is applied
+ *     to each document in turn, so adding a batch of documents can result in some being added and some
+ *     ignored, depending on what is already in the index. If all of the documents are skipped, no changes
+ *     to the index will occur.
+ * </p>
+ * <p>
+ *     These two forms of skipping can be switched on or off independently, by using init params:
+ * </p>
+ * <ul>
+ *     <li><code>skipInsertIfExists</code> - This boolean parameter defaults to
+ *          <code>true</code>, but if set to <code>false</code> then inserts (i.e. not Atomic Updates)
+ *          will be passed through unchanged even if the document already exists.</li>
+ *     <li><code>skipUpdateIfMissing</code> - This boolean parameter defaults to
+ *         <code>true</code>, but if set to <code>false</code> then Atomic Updates
+ *          will be passed through unchanged regardless of whether the document exists.</li>
+ * </ul>
+ * <p>
+ *     This implementation is a simpler alternative to <code>DocBasedVersionConstraintsProcessorFactory</code>
+ *     when you are not concerned with versioning, and just want to quietly ignore duplicate documents and/or
+ *     silently skip updates to non-existent documents (in the same way a database UPDATE would).
+ * </p>
+ */
+public class SkipExistingDocumentsProcessorFactory extends UpdateRequestProcessorFactory implements SolrCoreAware, UpdateRequestProcessorFactory.RunAlways {
```

Comment:
A lot of the code in this file is taken from `DocBasedVersionConstraintsProcessorFactory` and simplified